torchvision/transforms/functional.py

```diff
@@ -406,6 +406,46 @@ def hflip(img):
 
     return img.transpose(Image.FLIP_LEFT_RIGHT)
 
+def _get_perspective_coeffs(startpoints, endpoints):
+    """Helper function to get the coefficients (a, b, c, d, e, f, g, h) for the perspective transforms.
+    
+    In Perspective Transform each pixel (x, y) in the orignal image gets transformed as, 
+     (x, y) -> ( (ax + by + c) / (gx + hy + 1), (dx + ey + f) / (gx + hy + 1) )
+
+    Args:
+        List containing [top-left, top-right, bottom-right, bottom-left] of the orignal image,
+        List containing [top-left, top-right, bottom-right, bottom-left] of the transformed
+                    image
+    Returns:
+        octuple (a, b, c, d, e, f, g, h) for transforming each pixel.
+    """
+    matrix = []
+
+    for p1, p2 in zip(endpoints, startpoints):
+        matrix.append([p1[0], p1[1], 1, 0, 0, 0, -p2[0] * p1[0], -p2[0] * p1[1]])
+        matrix.append([0, 0, 0, p1[0], p1[1], 1, -p2[1] * p1[0], -p2[1] * p1[1]])
+
+    A = torch.tensor(matrix, dtype=torch.float)
+    B = torch.tensor(startpoints, dtype=torch.float).view(8)
+    res = torch.gels(B, A)[0]
+    return res.squeeze_(1).tolist()
+
+def perspective(img, coeffs, interpolation=Image.BICUBIC):
```

Comment:
Can we make `coeffs` be `startpoints` / `endpoints` ?
The way I see the functional interface, it's not just a wrapper of PIL, but instead the `torchvision` transforms are a wrapper around the functional transforms (with potentially some randomness added).

Could you move `get_perspective_coeffs` inside here?